File: Lecture 9 Introduction to Arrays in C++/ReverseArray.cpp
#include <iostream>
using namespace std;

void printReverseArray(int arr[],int size){

for (int i = size-1; i >=0; i--)
{
    cout << arr[i] << " ";
}

}
void printArray(int arr[],int size){

for (int i = 0; i <size; i++)
{
    cout << arr[i] << " ";
}

}
int main(){

   int size;
   cin >> size;
   int array[10];
   for (int i=0 ; i<size ; i++ )
   {
    cin >> array[i];
   }
   // Print original array
   cout << "Original Array:" << endl;
   printArray(array,size);

cout << "reverse array : "<< endl;
    
    printReverseArray(array,size);

    
    
   
}

File: Java/Lec 8-Introduction to Arrays and ArrayList in Java/code/ReverseArray.java
package code;

public class ReverseArray {
    public static void main(String[] args) {
        int[] array = {7,8,9};
        System.out.println("Original Array:");
        printArray(array);

        
        reverseArray(array);
        System.out.println("\nReversed Array:");
        printArray(array);

    }
static void printArray(int[] array) {
    for (int i = 0; i < array.length; i++) {
        System.out.print(array[i] + "  ");
    }
}

    static void reverseArray(int[] array) {
        int start = 0;
        int end = array.length-1;
        while (start < end) {
            // Swapping the elements
            swap(array, start, end);
        start++;
        end--;
        }


    }
    static void swap(int[] arr, int a, int b){
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;

    }
}
